proto/groups/v1alpha/groups.proto

```diff
@@ -0,0 +1,197 @@
+
+package com.mintter.groups.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "mintter/backend/genproto/groups/v1alpha;groups";
+
+// Groups service exposes the functionality for managing group entities.
+// This API uses fully-qualified EID as group IDs in all the requests.
+service Groups {
+  // Creates a new group.
+  rpc CreateGroup(CreateGroupRequest) returns (Group);
+
+  // Gets a group by ID.
+  rpc GetGroup(GetGroupRequest) returns (Group);
+
+  // Updates a group.
+  rpc UpdateGroup(UpdateGroupRequest) returns (Group);
+
+  // Lists members of a group.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Lists content of a group.
+  rpc ListContent(ListContentRequest) returns (ListContentResponse);
+
+  // Lists groups.
+  rpc ListGroups(ListGroupsRequest) returns (ListGroupsResponse);
+}
+
+// Request to create a group.
+message CreateGroupRequest {
+  // Required. Title of the new group.
+  string title = 1;
+
+  // Optional. Description of the new group.
+  string description = 2;
+
+  // Optional. List of initial members for the new group.
+  // Members can also be managed with separate requests after group is already created.
+  map<string, Role> members = 3;
+}
+
+// Request to get a group.
+message GetGroupRequest {
+  // Required. ID of the group to get.
+  // Must be a fully-qualified EID.
+  string id = 1;
+}
+
+// Request to update a group.
+// Some fields are required, even if unchanged,
+// to distinguish between setting the value to an empty string, from an unchanged value.
+// Read the comments!
+message UpdateGroupRequest {
+  // ID of the group to update.
+  string id = 1;
+
+  // Required. Title of the Group.
+  // Must always be provided, even if unchanged.
+  string title = 2;
+
+  // Required. Description of the Group. Can be empty string.
+  // Must always be provided, even if unchanged.
+  string description = 3;
+
+  // Optional. List of members to be updated in the Group.
+  // Key is the member's Account ID,
+  // value is the Role.
+  // To remove a member from the group, set the role to unspecified.
+  map<string, Role> updated_members = 4;
+
+  // Optional. List of content to be updated in the Group.
+  // Key is a pretty path on which the content is published,
+  // value is a Hyperdocs URL of the content.
+  // To unpublish content set the pretty path to an empty string.
+  map<string, string> updated_content = 5;
+}
+
+// Response to list members.
+message ListMembersRequest {
+  // Required. ID of the group to list members of.
+  string id = 1;
+
+  // Optional. Version of the group to list members of.
+  // If not specified, the latest version of the group
+  // according to its members is used.
+  string version = 2;
+
+  // Optional. Maximum number of members to return.
+  int32 page_size = 3;
+
+  // Optional. Page token to continue listing members from.
+  string page_token = 4;
+}
+
+// Response to list members.
+message ListMembersResponse {
```

Comment:
@ericvicenti This is a response type. The request does take the group ID and a version. But not sure response needs that: the caller should already know what they asked for, right? If you think it's needed in the response as well, I can add it ofc.

Reply:
Oops, I see! Eventually `ListMembersRequest` may need a parameter that allows you to see the members of the site version, vs the members of the local version

Reply:
ListMembersRequest already defines optional version argument. This could be any version, including the site's one (which we'll be able to see in GetGroup call I guess).